astronomer/providers/sftp/hooks/sftp.py

```diff
@@ -60,12 +50,38 @@ def __init__(  # nosec: B107
         self.port = port
         self.username = username
         self.password = password
-        self.known_hosts = known_hosts
+        self.known_hosts: bytes | str = os.path.expanduser(known_hosts)
         self.key_file = key_file
         self.passphrase = passphrase
         self.private_key = private_key
-        self.no_host_key_check = True
-        self.host_key = None
+
+    def _parse_extras(self, conn: Connection) -> None:
+        """Parse extra fields from the connection into instance fields"""
+        extra_options = conn.extra_dejson
+        if "key_file" in extra_options and self.key_file == "":
+            self.key_file = extra_options["key_file"]
+        if "known_hosts" in extra_options:
+            self.known_hosts = extra_options["known_hosts"]
+        if ("passphrase" or "private_key_passphrase") in extra_options:
+            self.passphrase = extra_options["passphrase"]
+        if "private_key" in extra_options:
+            self.private_key = extra_options["private_key"]
+
+        host_key = extra_options.get("host_key")
+        no_host_key_check = extra_options.get("no_host_key_check")
+
+        if no_host_key_check is not None:
+            no_host_key_check = str(no_host_key_check).lower() == "true"
+            if host_key is not None and no_host_key_check:
```

Comment:
I'm confused how ~host_key` and `known_hosts` interact -- what is the difference between them? Why do we need two options?

Reply:
If someone is being extra careful and sets `no_host_key_check=False`, this will require host_key to be set, by-and not let the connection use the value given in known_hosts.

Reply:
We discussed this over a call and verified the scenario is working well.